### PR TITLE
Matter.framework should explicitly look for things in the SRP domain.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
The operational browse and connectivity monitoring in Matter.framework should happen on both the SRP domain and the default domain, so we are sure we'll notice Thread devices properly.

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

File: src/darwin/Framework/CHIP/MTROperationalBrowser.mm [hunk 1]
```diff
@@ -43,23 +46,30 @@
...
-        ChipLogError(Controller, "Failed to set up dispatch queue properly");
+        ChipLogError(Controller, "Failed to set up dispatch queue properly for persistent operational browse: %" PRId32, err);
         DNSServiceRefDeallocate(mBrowseRef);
         return;
     }
 
     mInitialized = true;
+
+    for (auto domain : kBrowseDomains) {
+        auto browseRef = mBrowseRef; // Mandatory copy because of kDNSServiceFlagsShareConnection.
```
Comment (jtung-apple) [inline src/darwin/Framework/CHIP/MTROperationalBrowser.mm:67]: My understanding is that the DNSServiceRefs copied/created from a shared connection also need to have `DNSServiceRefDeallocate` called separately when they are done, or we'll be leaking some memory.

Comment (bzbarsky-apple) [reply src/darwin/Framework/CHIP/MTROperationalBrowser.mm:67]: Hmm.  We are very much not doing that in our src/platform code.  My understanding was that when the shared connection is closed, all the resources associated with it are cleaned up.  But let me double-check that.

**Review by yunhanw-google — APPROVED** [review]

File: src/darwin/Framework/CHIP/MTROperationalBrowser.mm (hunk 1 above)
Comment (bzbarsky-apple) [reply src/darwin/Framework/CHIP/MTROperationalBrowser.mm:67]: So https://developer.apple.com/documentation/dnssd/1823436-anonymous/kdnsserviceflagsshareconnection says:

> Calling DNSServiceRefDeallocate(ref) for a particular operation's DNSServiceRef terminates just that operation. Calling DNSServiceRefDeallocate(ref) for the main shared DNSServiceRef (the parent DNSServiceRef, originally created by DNSServiceCreateConnection(&ref)) automatically terminates the shared connection and all operations that were still using it. After doing this, DO NOT then attempt to deallocate any remaining subordinate DNSServiceRef's. The memory used by those subordinate DNSServiceRef's has already been freed, so any attempt to do a DNSServiceRefDeallocate (or any other operation) on them will result in accesses to freed memory, leading to crashes or other equally undesirable results.

So yes, a least as per the API documentation what we have here is fine: once we deallocate `mBrowseRef` all the things go away.

**Comment by bzbarsky-apple** [comment]
CI failure is real, looking into what we should do there.

**Comment by bzbarsky-apple** [comment]
Fast-tracking platform-specific change with platform owner review.

**Pull request merged**